code/backend/batch/utilities/search/Search.py

```diff
@@ -19,7 +19,11 @@ def get_search_handler(env_helper: EnvHelper):
     def get_source_documents(search_handler, question):
         if isinstance(search_handler, IntegratedVectorizationSearchHandler):
             search_results = search_handler.query_search(question)
-            return Search.generate_source_documents(search_results)
+            return (
+                Search.generate_source_documents(search_results)
+                if search_results
+                else []
```

Comment:
Are we missing a test case for this "else" condition?

Reply:
Added